test/heterarchy.spec.coffee

```diff
@@ -141,20 +156,54 @@ describe 'heterarchy', ->
                 EditableScrollablePane, ScrollablePane, EditablePane,
                 Pane, ScrollingMixin, EditingMixin, Object ]
 
+
     describe 'multi', ->
 
-        it 'calls super properly in multi case', ->
-            obj = new D
-            (mro D).should.eql [D, B, C, A, Object]
-            obj.method().should.equal "D>B>C>A"
+        describe 'instance methods', ->
+
+            it 'calls super properly in multi case', ->
+                obj = new D()
+                (mro D).should.eql [D, B, C, A, Object]
+                obj.method().should.equal 'D>B>C>A'
+
+            it 'calls super properly in recursive multi case', ->
+                obj = new G()
+                (mro G).should.eql [G, D, B, F, C, E, A, Object]
+                obj.method().should.equal 'G>D>B>F>C>E>A'
+
+
+        describe 'class methods', ->
+
+            it 'calls super properly in multi case', ->
+                D.method().should.equal 'D>B>C>A'
+
+            it 'calls super properly in recursive multi case', ->
+                # method is overridden
+                G.method().should.equal 'G>D>B>F>C>E>A'
 
-        it 'calls super properly in recursive multi case', ->
-            obj = new G
-            (mro G).should.eql [G, D, B, F, C, E, A, Object]
-            obj.method().should.equal "G>D>B>F>C>E>A"
+                # closure for not overwriting the value of e.g. `A`
+                ((A, B, C) ->
+                    # method is not overridden
+                    class A
+                        @classMethod: () ->
+                            return super() + 'Base1'
+
+                    class B
+                        @classMethod: () ->
+                            return 'Base2'
+
+                    class C extends multi A, B
+
+                    C.classMethod().should.equal 'Base2Base1'
+                )(0, 0, 0)
```

Comment:
It actually does work with

```coffee
class A
  @a = 1

do (A) ->
  class A
    @a = 2

alert A.a # this alerts 1
```

Because the local `A` is being assigned. Thus it does not matter if the local `A` equals the outer-scope `A` - tricky ^^